exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinExporterConfiguration.java

```diff
@@ -0,0 +1,159 @@
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * Configurations for {@link ZipkinSpanExporter}.
+ *
+ * @since 0.4.0
+ */
+@AutoValue
+@Immutable
+public abstract class ZipkinExporterConfiguration {
+
+  ZipkinExporterConfiguration() {}
+
+  /**
+   * Returns the service name.
+   *
+   * @return the service name.
+   * @since 0.4.0
+   */
+  public abstract String getServiceName();
+
+  /**
+   * Returns the Zipkin V2 URL.
+   *
+   * @return the Zipkin V2 URL.
+   * @since 0.4.0
+   */
+  public abstract String getV2Url();
+
+  /**
+   * Returns the Zipkin sender.
+   *
+   * @return the Zipkin sender.
+   * @since 0.4.0
+   */
+  @Nullable
+  public abstract Sender getSender();
+
+  /**
+   * Returns the {@link SpanBytesEncoder}.
+   *
+   * <p>Default is {@link SpanBytesEncoder#JSON_V2}.
+   *
+   * @return the {@code SpanBytesEncoder}
+   * @since 0.4.0
+   */
+  public abstract SpanBytesEncoder getEncoder();
+
+  /**
+   * Returns a new {@link Builder}.
+   *
+   * @return a {@code Builder}.
+   * @since 0.4.0
+   */
+  public static Builder builder() {
+    return new AutoValue_ZipkinExporterConfiguration.Builder()
+        .setV2Url("")
+        .setEncoder(SpanBytesEncoder.JSON_V2);
+  }
+
+  /**
+   * Builder for {@link ZipkinExporterConfiguration}.
+   *
+   * @since 0.4.0
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    Builder() {}
+
+    /**
+     * Sets the service name.
+     *
+     * @param serviceName the service name.
+     * @return this Builder instance.
+     * @since 0.4.0
+     */
+    public abstract Builder setServiceName(String serviceName);
+
+    /**
+     * Sets the Zipkin V2 URL, e.g.: "http://127.0.0.1:9411/api/v2/spans".
+     *
+     * <p>At least one of {@code V2Url} and {@code Sender} needs to be specified. If both {@code
+     * V2Url} and {@code Sender} are set, {@code Sender} takes precedence.
+     *
+     * @param v2Url the Zipkin V2 URL.
+     * @return this Builder instance.
+     * @since 0.4.0
+     */
+    public abstract Builder setV2Url(String v2Url);
+
+    /**
+     * Sets the Zipkin sender.
+     *
+     * <p>At least one of {@code V2Url} and {@code Sender} needs to be specified. If both {@code
+     * V2Url} and {@code Sender} are set, {@code Sender} takes precedence.
+     *
+     * <p>Note: if you provide a {@link Sender} instance via this method, the {@link Sender#close()}
+     * method will be called when the exporter is shut down.
+     *
+     * @param sender the Zipkin sender.
+     * @return this Builder instance.
+     * @since 0.4.0
+     */
+    public abstract Builder setSender(Sender sender);
+
+    /**
+     * Sets the {@link SpanBytesEncoder}.
+     *
+     * @param encoder the {@code SpanBytesEncoder}.
+     * @return this Builder instance
+     * @since 0.4.0
+     */
+    public abstract Builder setEncoder(SpanBytesEncoder encoder);
```

Comment:
change to `BytesEncoder<Span>` as otherwise you'll break non-zipkin formats like https://github.com/openzipkin/zipkin-gcp/blob/master/sender-stackdriver/src/main/java/zipkin2/reporter/stackdriver/StackdriverEncoder.java

(ack that they would make their own exporter, but anyway no need to narrow the type)